src/evict/evict_lru.c

```diff
@@ -2927,6 +2927,26 @@ __wti_evict_app_assist_worker(
                 break;
         }
 
+        /* Update cache metrics if we are stalling to detect a timeout quicker.*/
```

Comment:
Overall this change is looking good, but I was wondering if it's possible to refactor this into a function as we're essentially repeating the same logic at line 2985? If the change is urgent, don't worry about it immediately and create a ticket for refactoring.

Reply:
Since we are only using part of the function I have not refactored.  [d61c357](https://github.com/wiredtiger/wiredtiger/pull/12037/commits/d61c357ecd36ecf575fe0d8ed6a2c3f1bf2a24e1)